domain/upgrade/state/state.go

```diff
@@ -172,198 +181,163 @@ func (st *State) StartUpgrade(ctx context.Context, upgradeUUID domainupgrade.UUI
 		return errors.Trace(err)
 	}
 
-	getUpgradeStartedQuery := `
-SELECT upgrade_state_type.id 
-FROM upgrade_info 
-	LEFT JOIN upgrade_state_type
-		ON upgrade_info.state_type_id = upgrade_state_type.id
-WHERE uuid = ?
-`
+	info := Info{
+		UUID: upgradeUUID.String(),
+	}
 
-	startUpgradeQuery := "UPDATE upgrade_info SET state_type_id = ? WHERE uuid = ? AND state_type_id = ?"
+	getUpgradeStartedStmt, err := st.Prepare(`
+SELECT &Info.state_type_id 
+FROM   upgrade_info 
+WHERE  uuid = $Info.uuid
+`, info)
```

Comment:
This is cool, if I understand it right.

I didn't know you could use the same struct as both the input and output. You fill in the uuid field, and then sqlair fills in the rest?

Reply:
Yeep :)